neuralmonkey/encoders/sequence_cnn_encoder.py

```diff
@@ -46,56 +49,70 @@ def __init__(self,
-                                             name="train_mode")
-
-            self.inputs = tf.placeholder(tf.int32,
-                                         shape=[None, None],
-                                         name="encoder_input")
-
-            self._input_mask = tf.placeholder(
-                tf.float32, shape=[None, None],
-                name="encoder_padding")
-
-            with tf.variable_scope("input_projection"):
-                self.embedding_matrix = tf.get_variable(
-                    "word_embeddings", [len(vocabulary), embedding_size],
-                    initializer=tf.random_normal_initializer(stddev=0.01))
-                embedded_inputs = dropout(
-                    tf.nn.embedding_lookup(self.embedding_matrix, self.inputs),
-                    dropout_keep_prob,
-                    self.train_mode)
-
-            pooled_outputs = []
-            for filter_size, num_filters in filters:
-                with tf.variable_scope("conv-maxpool-%s" % filter_size):
-                    # Convolution Layer
-                    filter_shape = [filter_size, embedding_size, num_filters]
-                    w_filter = tf.get_variable(
-                        "conv_W", filter_shape,
-                        initializer=tf.random_uniform_initializer(-0.5, 0.5))
-                    b_filter = tf.get_variable(
-                        "conv_bias", [num_filters],
-                        initializer=tf.constant_initializer(0.0))
-                    conv = tf.nn.conv1d(
-                        embedded_inputs,
-                        w_filter,
-                        stride=1,
-                        padding="VALID",
-                        name="conv")
-
-                    # Apply nonlinearity
-                    conv_relu = tf.nn.relu(tf.nn.bias_add(conv, b_filter))
-
-                    # Max-pooling over the outputs
-                    pooled = tf.reduce_max(conv_relu, 1)
-                    pooled_outputs.append(pooled)
-
-            # Combine all the pooled features
-            self.encoded = tf.concat(pooled_outputs, axis=1)
-    # pylint: enable=too-many-arguments,too-many-locals
+        self._embedding_size = embedding_size
+        self._dropout_keep_prob = dropout_keep_prob
+        self._filters = filters
+
+    # pylint: disable=no-self-use
+    @tensor
+    def train_mode(self) -> tf.Tensor:
+        return tf.placeholder(tf.bool, shape=[],
+                              name="train_mode")
+
+    @tensor
+    def inputs(self) -> tf.Tensor:
+        return tf.placeholder(tf.int32,
+                              shape=[None, None],
+                              name="encoder_input")
+
+    @tensor
+    def _input_mask(self) -> tf.Tensor:
+        return tf.placeholder(tf.float32, shape=[None, None],
+                              name="encoder_padding")
+    # pylint: enable=no-self-use
+
+    @tensor
+    def _embedded_inputs(self) -> tf.Tensor:
```

Comment:
No embedování bude speciální třída, takže je jedno, jak se to udělá tady, protože odsud se to stejně smaže.